src/puppetlabs/ezbake/core.clj

```diff
@@ -699,6 +707,38 @@ Additional uberjar dependencies:
 (defmulti action
   (fn [action & args] action))
 
+(defn munge-deps
+  ([deps]
+   (into {} (concat (for [[k v] deps]
+                      (let [[project-name version] k
+                            dependencies (munge-deps v)
+                            metadata (if (empty? dependencies) {"version" version} {"version" version "dependencies" dependencies })
```

Comment:
My Clojure is rusty, and collapsing code isn't always a good idea, but a couple things you might consider here:
- Moving the `let` block into the `for` using the convenience `:let` form
- Moving the `[project-name version]` destructuring into the `[k v]` since you don't ever need `k` itself, so something like `(for [[[project-name version] v] deps]...`
- Rename `v` to match what it represents, probably just something like `deps` or `dependencies`
- Insert some newlines into the `metadata` line so it's not trailing off the screen
- The call to `(str project-name)` at the bottom is ambiguous since `str` is the name of the `clojure.string` library import; this should probably be explicit